### DISCUSSION TITLE
Method Illuminate\Database\Eloquent\Collection::user does not exist.

### DISCUSSION THREAD

**Discussion by Trevorkeya** [rappasoft/laravel-livewire-tables#1081] [Q&A]
I am trying to show all posts of a user but in trying to pass an if statement I'm getting the above error 
here is my blade file:

 @if(Auth::user()->id == $county->user()->id)
        @foreach($county as $county)
            <div class="county"><a  href="{{ route('show', $county->id) }}" class="btn">
                <h5>{{$county->title}}</h5><br>
                {{$county->body}}
                <br>
                {{date('Y-m-d ', strtotime($county->created_at)) }}
            </a>
            </div>
            <br>
        
        @endforeach
        @endif

here is my profile controller;
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use App\Models\User;
use App\Models\county;
use App\Http\Controllers\Controller;
use Illuminate\support\facades\Auth;

class ProfileController extends Controller
{
    public function index()
    {
        $county = county::all();
        $user = User::all();
        return view('profile.index', compact('county'));
 
    }

    public function edit()
    {
        return view('profile.profile');
    }

    public function updateUserDetails(Request $request)
    {
        $request->validate([
            'username' => ['required','string'],
            'phone' => ['required','digits:10'],
            'status' => ['required','string'],
            'organization' => ['required','string'],
            'image' => 'required|image',

    
        ]);
    
        $user = User::findOrFail(Auth::user()->id);
        $user->update([
            'name' => $request->username
        ]);
    
        $user->profile()->updateOrCreate(
            [
                'user_id' => $user->id,
            ],
            [
                'phone' => $request->phone,
                'status' =>$request->status,
                'organization' => $request->organization,
                'image' => $request->image,
            ]
        );
        $file=$request->image;
    
        $imageName = time().'.'.$request->image->extension();
        $request->image->move(public_path('images'), $imageName);  
        
        $user->image=$imageName;
    
        return redirect('/profile')->with('success','User Profile Updated Successfully!');
    }
}

here is my User model
<?php

namespace App\Models;

// use Illuminate\Contracts\Auth\MustVerifyEmail;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Illuminate\Notifications\Notifiable;
use Laravel\Sanctum\HasApiTokens;
use Illuminate\Contracts\Auth\MustVerifyEmail;
use App\Models\profile;
use App\Models\county;

class User extends Authenticatable
{
    use HasApiTokens, HasFactory, Notifiable;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'name',
        'email',
        'password',
        'role_as',
    ];

    public function profile()
    {
        return $this->hasOne(Profile::class, 'user_id','id');
    }

    public function county()
    {
        return $this->hasMany(county::class);
    }
    /**
     * The attributes that should be hidden for serialization.
     *
     * @var array<int, string>
     */
    protected $hidden = [
        'password',
        'remember_token',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'email_verified_at' => 'datetime',
    ];

   
}

heres my county model

<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class county extends Model
{
    use HasFactory;
    protected $table = 'counties';
    protected $fillable =[
        'user_id',
        'title',
        'pdf',
        'body',
    ];

   public function user()
   {
    return $this->belongsTo(User::class);
   }
}

**Comment by lrljoe** [comment]
Use a where clause to only load the right records rather than using if statements.

If you post your full component then I can give you a steer .

Please use the code tags so that it keeps its formatting.
